src/NLog.Extensions.Hosting/Extensions/ConfigureExtensions.cs

```diff
@@ -41,7 +41,17 @@ public static IHostBuilder UseNLog(this IHostBuilder builder, NLogProviderOption
                 ConfigurationItemFactory.Default.RegisterItemsFromAssembly(typeof(ConfigureExtensions).GetTypeInfo()
                     .Assembly);
 
-                services.AddSingleton(new LoggerFactory().AddNLog(options));
+                services.AddSingleton<ILoggerProvider>(serviceProvider =>
+                {
+                    var provider = new NLogLoggerProvider(options ?? new NLogProviderOptions());
+                    if (hostbuilder.Configuration != null)
+                    {
+                        // TODO ConfigSettingLayoutRenderer.DefaultConfiguration = hostbuilder.Configuration;
```

Comment:
when should be do this TODO?

Reply:
Next PR. You just requested me to split this into more PRs